shell/initialize/client.js

```diff
@@ -505,108 +478,6 @@ function nuxtReady(_app) {
   }
 }
 
-const noopData = () => {
-  return {};
-};
-const noopFetch = () => {};
-
-// Special hot reload with asyncData(context)
-function getNuxtChildComponents($parent, $components = []) {
-  $parent.$children.forEach(($child) => {
-    if ($child.$vnode && $child.$vnode.data.nuxtChild && !$components.find((c) => (c.$options.__file === $child.$options.__file))) {
-      $components.push($child);
-    }
-    if ($child.$children && $child.$children.length) {
-      getNuxtChildComponents($child, $components);
-    }
-  });
-
-  return $components;
-}
-
-function hotReloadAPI(_app) {
-  if (!module.hot) {
-    return;
-  }
-
-  const $components = getNuxtChildComponents(window.$globalApp, []);
-
-  $components.forEach(addHotReload.bind(_app));
-}
-
-function addHotReload($component, depth) {
-  if ($component.$vnode.data._hasHotReload) {
-    return;
-  }
-  $component.$vnode.data._hasHotReload = true;
-
-  const _forceUpdate = $component.$forceUpdate.bind($component.$parent);
-
-  $component.$vnode.context.$forceUpdate = async() => {
-    const Components = getMatchedComponents(router.currentRoute);
-    let Component = Components[depth];
-
-    if (!Component) {
-      return _forceUpdate();
-    }
-    if (typeof Component === 'object' && !Component.options) {
-      // Updated via vue-router resolveAsyncComponents()
-      Component = Vue.extend(Component);
-      Component._Ctor = Component;
-    }
-    this.error();
-    const promises = [];
-    const next = function(path) {
-      this.$loading.finish && this.$loading.finish();
```

Comment:
This makes me wonder if $loading can be removed entirely. I didn't experience any regressions while clicking through the app in dev.